### PR TITLE
Move Nimbus startup into Application Services

### PR DESCRIPTION
**Pull request by jhugman**
This builds on the work of https://github.com/mozilla-mobile/fenix/pull/27650 and https://github.com/mozilla-mobile/focus-android/pull/8012 , pulling them together into a common codebase.

Part of fixing [EXP-2893](https://mozilla-hub.atlassian.net/browse/EXP-2893).

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[ff-android: firefox-android-branch-name]` and/or `[fenix: fenix-branch-name]` to the PR title.

### CODE REVIEW

File: components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+     * A flag to indicate if this is being run on the first run of the app. This is used to control
+     * whether the `initial_experiments` file is used to populate Nimbus.
+     */
+    var isFirstRun: Boolean = true
+
+    /**
+     * A optional raw resource of a file downloaded at or near build time from Remote Settings.
+     */
+    @RawRes
+    var initialExperiments: Int? = null
```
Comment (jeddai) [inline components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusBuilder.kt:48]: I wonder if converting this to be a String type would be helpful as we plan to integrate Nimbus into other apps. We could have other Kotlin apps that aren't Android, and as far as I'm aware the raw resource int thing is an Android thing.

If it's the string value of the initial experiments, I think that also lends itself a bit better to testing and extensibility.

**Review by jeddai — COMMENTED** [review]
Looks solid, I've got one question.

File: components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusBuilder.kt (hunk 1 above)
Comment (jhugman) [reply components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusBuilder.kt:48]: I think at this point, building out a non-Android Kotlin version would be a quite a large undertaking (for example, we depend on `android.content.Context`), for currently not much demand.

I think I'm inclined to say "Let's cross that bridge when we get to it".

Comment (jeddai) [reply components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusBuilder.kt:48]: Sounds good to me!

**Review by jeddai — APPROVED** [review]

**Pull request merged**